_episodes_rmd/04-r-data-cleaning-merging.Rmd

```diff
@@ -31,151 +32,167 @@ knitr_fig_path("04-")
+
 {.callout}
 
 # Joining data frames
 
 [*Back to top*](#contents)
 
 
-Now we're ready to join our CO2 emissions data to the gapminder data. Previously we saw that we could read in and filter the gapminder data like this to get the data from the Americas for 2007 so we can create a new dataframe with our filtered data:
+Now we're almost ready to join our pollution data to the smoking and lung cancer data. Previously we saw that we could read in and filter the smoking and lung cancer data to get the data from 2019 to create a new dataframe with our filtered data:
 
 ```{r LoadGapminder2007}
-gapminder_data_2007 <- read_csv("data/gapminder_data.csv") %>%
-  filter(year == 2007 & continent == "Americas") %>%
-  select(-year, -continent)
+smoking_2019 <- read_csv("data/smoking_cancer.csv") %>%
+  filter(year == 2019) %>%
+  select(-year)
 ```
 
-Look at the data in `co2_emissions` and `gapminder_data_2007`. If you had to merge these two data frames together, which column would you use to merge them together? If you said "country" - good job!
+Look at the data in `pollution_2019` and `smoking_2019`. If you had to merge these two data frames together, which columns would you use to merge them together? If you said `location_name` and `country`, you're right! But before we join the datasets, we need to make sure these columns are named the same thing.
 
-We'll call country our "key". Now, when we join them together, can you think of any problems we might run into when we merge things? We might not have CO2 emissions data for all of the countries in the gapminder dataset and vice versa. Also, a country might be represented in both data frames but not by the same name in both places. As an example, write down the name of the country that the University of Michigan is in - we'll come back to your answer shortly!
+> ## Review: Re-naming columns
+> Rename the `location_name` column to `country` in the `pollution_2019` dataset. Go ahead and overwrite and store in the `pollution_2019` object
+> 
+> > ## Solution: 
+> > ```{r isna}
+> > 
+> > pollution_2019 <- pollution_2019 %>%
+> >   rename(country = location_name) 
+> > ```
+> > 
+> > Note that the column is labeled `country` even though it has values beyond the names of countries. We will take care of this later when we join datasets.
+> {: .solution}
+{: .challenge}
+
+Because the column is now present in both datasets, we'll call country our "key". Now, when we join them together, can you think of any problems we might run into when we merge things? We might not have pollution data for all of the countries in the `smoking_2019` dataset and vice versa. Also, a country might be represented in both data frames but not by the same name in both places.
 
 The dplyr package has a number of tools for joining data frames together depending on what we want to do with the rows of the data of countries that are not represented in both data frames. Here we'll be using `inner_join()` and `anti_join()`. 
```

Comment:
I don't know about you, but I almost always use `full_join()` or `left_join()` rather than the ones explained here. What do you think about changing it to these? As I mentioned above it would also make it a easier to discuss missing data as well.

Because what we're really interested in is keeping all the data from the smoking dataset and adding in the relevant rows from the air pollution dataset, right?

We could still use `anti_join()` to check that everything went okay at the end if we want.